src/monsters.cpp

```diff
@@ -186,6 +239,175 @@ void MonsterType::createLoot(Container* corpse)
+		root->outfit.lookMount = child->outfit.lookMount;
+		root->lookcorpse = child->lookcorpse;
+
+		root->skull = child->skull;
+		root->conditionImmunities = child->conditionImmunities;
+		root->damageImmunities = child->damageImmunities;
+		root->race = child->race;
+		root->isSummonable = child->isSummonable;
+		root->isIllusionable = child->isIllusionable;
+		root->isConvinceable = child->isConvinceable;
+		root->isAttackable = child->isAttackable;
+		root->isHostile = child->isHostile;
+		root->lightLevel = child->lightLevel;
+		root->lightColor = child->lightColor;
+		root->manaCost = child->manaCost;
+
+		root->elementMap[COMBAT_PHYSICALDAMAGE] = child->elementMap[COMBAT_PHYSICALDAMAGE];
+		root->elementMap[COMBAT_ICEDAMAGE] = child->elementMap[COMBAT_ICEDAMAGE];
+		root->elementMap[COMBAT_EARTHDAMAGE] = child->elementMap[COMBAT_EARTHDAMAGE];
+		root->elementMap[COMBAT_FIREDAMAGE] = child->elementMap[COMBAT_FIREDAMAGE];
+		root->elementMap[COMBAT_ENERGYDAMAGE] = child->elementMap[COMBAT_ENERGYDAMAGE];
+		root->elementMap[COMBAT_HOLYDAMAGE] = child->elementMap[COMBAT_HOLYDAMAGE];
+		root->elementMap[COMBAT_DEATHDAMAGE] = child->elementMap[COMBAT_DEATHDAMAGE];
+		root->elementMap[COMBAT_DROWNDAMAGE] = child->elementMap[COMBAT_DROWNDAMAGE];
+		root->elementMap[COMBAT_LIFEDRAIN] = child->elementMap[COMBAT_LIFEDRAIN];
+		root->elementMap[COMBAT_MANADRAIN] = child->elementMap[COMBAT_MANADRAIN];
+
+		if (!child->voiceVector.empty()) {
+			for (voiceBlock_t voices : child->voiceVector) {
+				root->voiceVector.push_back(voices);
+			}
+		}
+
+		if (!child->summonList.empty()) {
+			for (summonBlock_t summons : child->summonList) {
+				root->summonList.push_back(summons);
+			}
+		}
+
+		if (!child->lootItems.empty()) {
+			for (LootBlock loot : child->lootItems) {
+				root->lootItems.push_back(loot);
+			}
+		}
+
+	} else {
+		std::cout << "[Error - MonsterType::clone] unknown type" << std::endl;
+	}
+}
+
+void MonsterType::loadLoot(MonsterType* monsterType, Loot* loot) 
+{
+	LootBlock lootblock;
+	
+	lootblock.id = loot->id;
+	lootblock.countmax = loot->countMax;
+	
+	if (loot->chance != 0) {
+		lootblock.chance = loot->chance;
+	} else {
+		lootblock.chance = MAX_LOOTCHANCE;
+	}
+	
+	if (loot->actionId != -1) {
+		lootblock.actionId = loot->actionId;
+	}
+
+	if (loot->subType != -1) {
+		lootblock.subType = loot->subType;
+	} else {
+		uint32_t charges = Item::items[lootblock.id].charges;
+		if (charges != 0) {
+			lootblock.subType = charges;
+		}
+	}
+
+	if (!loot->text.empty()) {
+		lootblock.text = loot->text;
+	}
+
+	if (!loot->children.empty()) {
+		bool isContainer = Item::items[lootblock.id].isContainer();
+		if (isContainer) {
+			for (Loot* l : loot->children) {
+				loadChildLoot(l, lootblock);
+			}
+		}
+		monsterType->lootItems.push_back(lootblock);
+	} else {
+		monsterType->lootItems.push_back(lootblock);
+	}
+}
+
+void MonsterType::loadChildLoot(Loot* loot, LootBlock& parent)
```

Comment:
I already had that in mind, I just wanted to do that once everything was functional but thanks for reminding me.